mmv1/products/firebasehosting/CustomDomain.yaml

```diff
@@ -0,0 +1,510 @@
+      redirect_target: destination.domain.com
+    test_env_vars:
+      project_id: :PROJECT_NAME
+    test_vars_overrides:
+      # Don't add random suffix
+      custom_domain: '"full.source.domain.com"'
+      redirect_target: '"destination.domain.com"'
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firebasehosting_customdomain_cloud_run"
+    min_version: "beta"
+    primary_resource_id: "default"
+    vars:
+      site_id: site-id
+      custom_domain: run.custom.domain.com
+      cloud_run_service_id: cloud-run-service-via-hosting
+    test_env_vars:
+      project_id: :PROJECT_NAME
+    test_vars_overrides:
+      # Don't add random suffix
+      custom_domain: '"run.custom.domain.com"'
+parameters:
+  - !ruby/object:Api::Type::String
+    name: site_id
+    description: |
+      The ID of the site in which to create this custom domain association.
+    immutable: true
+    required: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: custom_domain
+    description: |
+      The ID of the `CustomDomain`, which is the domain name you'd like to use with Firebase Hosting.
+    immutable: true
+    required: true
+    url_param_only: true
+virtual_fields:
+  - !ruby/object:Api::Type::Boolean
+    name: wait_dns_verification
+    description: |
+      If true, Terraform will wait for DNS records to be fully resolved on the `CustomDomain`.
+      If false, Terraform will not wait for DNS records on the `CustomDomain`. Any issues in
+      the `CustomDomain` will be returned and stored in the Terraform state.
+    immutable: true
+    default_value: false
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    pattern: projects/{{project}}/sites/{{site_id}}/customDomains/{{custom_domain}}
+    description: |
+      The fully-qualified name of the `CustomDomain`.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: createTime
+    output: true
+    description: The `CustomDomain`'s create time.
+  - !ruby/object:Api::Type::Time
+    name: updateTime
+    output: true
+    description: The last time the `CustomDomain` was updated.
+  - !ruby/object:Api::Type::Time
+    name: deleteTime
+    output: true
+    description: |
+      The time the `CustomDomain` was deleted; null for `CustomDomains` that
+      haven't been deleted. Deleted `CustomDomains` persist for approximately 30
+      days, after which time Hosting removes them completely.
+  - !ruby/object:Api::Type::Time
+    name: expireTime
+    output: true
+    description: |
+      The minimum time before a soft-deleted `CustomDomain` is completely removed
+      from Hosting; null for `CustomDomains` that haven't been deleted.
+  - !ruby/object:Api::Type::Fingerprint
+    name: etag
+    output: true
+    description: |
+      A string that represents the current state of the `CustomDomain` and
+      allows you to confirm its initial state in requests that would modify it.
+  - !ruby/object:Api::Type::Enum
```

Comment:
Nit: I didn't mention this before, but you may want to consider changing these output-only enums to simple strings. The validation shouldn't be necessary, and a string would be more flexible to server-side changes.

Reply:
Done.